### ISSUE TITLE
Uncaught TypeError: Cannot read property 'toCacheKey' of undefined in TypeScript version

### ISSUE THREAD

**Issue by Fudoshiki** [turbolinks/turbolinks#447]
Version 5.3.0-beta.0

When visit any page

![2019-02-16 14 16 09](https://user-images.githubusercontent.com/4083992/52897372-73240600-31f5-11e9-845d-2a653e17b360.png)

**Comment by domchristie** [comment]
I'm not seeing this on [this demo](https://secret-bayou-73715.herokuapp.com/) (which also uses the latest version). Are you able to reproduce with a minimal app?

**Comment by Fudoshiki** [comment]
Try add document.addEventListener 'turbolinks:before-cache', ...

**Comment by domchristie** [comment]
> Try add document.addEventListener 'turbolinks:before-cache', ...

Hmm. I'm still not seeing this error :/ If you could link to an app which demonstrates this problem that'd be really helpful. Thanks.

**Comment by sstephenson** [comment]
Is it possible that another script in your application is calling private Turbolinks methods?

**Comment by Fudoshiki** [comment]
I will add example app

**Comment by Fudoshiki** [comment]
**app.coffee**
```coffee
#= require rails-ujs
#= require custom/turbolinks
#= require action_cable
#= require local-time
```

application.slim
```slim
doctype html
html
  head
    == javascript_include_tag 'app', async: true, 'data-turbolinks-track': true
```

Error with `async: true`

**Comment by rsmything** [comment]
The same problem, it appears if script tag was added with async="async" option

<script src="/assets/turbolinks.debug-4578b47429a4e47d0db555c338192a746ecd846a48f59b523ee4b3d261e25562.js" data-turbolinks-track="true" async="async"></script>

```
turbolinks.source.js:599 Uncaught TypeError: Cannot read property 'toCacheKey' of undefined
    at SnapshotCache.write (turbolinks.source.js:599)
    at SnapshotCache.put (turbolinks.source.js:591)
    at turbolinks.source.js:1415
```

**Comment by Fudoshiki** [comment]
Turbolinks 5.2.0 (coffee) works fine with `async: true`

**Comment by sstephenson** [comment]
Thank you!

**Comment by sstephenson** [comment]
This is fixed in [5.3.0-beta.1](https://github.com/turbolinks/turbolinks/releases/tag/v5.3.0-beta.1).
